### PR TITLE
[FP-09] Customer Satisfaction Performance

### PR DESCRIPTION
**Pull request by arturolinares24**
@rotorrest @jkahnc Ready for review

### CODE REVIEW

**Review by jkahnc — COMMENTED** [review]
just an observation about the return in the plot functions on `paths/customer_satisfaction_performance.py`. It should be `return True`.  That's all

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        """
+        self.shimoku.plt.html(
+            html=create_title_name_head(
+                title="Customer Satisfaction Performance",
+                subtitle="",
+            ),
+            order=self.order,
+        )
+        self.order += 1
+
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:56]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        self.plot_kpi_indicators()
+        self.plot_customer_satisfaction()
+        self.plot_orders_satisfaction()
+        self.plot_revenue_indicators()
+        self.plot_orders_returned_orders()
+        self.plot_revenue_analysis()
+
+    def plot_header(self):
+        """
+        Plot the header section of the dashboard.
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:46]: Add return to docstring

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                title="Customer Satisfaction Performance",
+                subtitle="",
+            ),
+            order=self.order,
+        )
+        self.order += 1
+
+    def plot_kpi_indicators(self):
+        """
+        Plot the Key Performance Indicators (KPI) section of the dashboard.
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:59]: Add return to docstring

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            cols_size=3,
+            value=orders_with_returns,
+            order=self.order,
+            rows_size=1,
+            name="Orders with Returns",
+            is_percentage=True,
+            color=orders_with_returns_status,
+        )
+        self.order += 1
+
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:115]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+    def plot_customer_satisfaction(self):
+        """
+        Plot the Customer Satisfaction section of the dashboard.
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:118]: Add return to docstring

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+            data=customer_satisfaction,
+            values="value",
+            names="status",
+            order=self.order,
+            rows_size=3,
+            cols_size=4,
+            title="Customer Satisfaction",
+        )
+        self.order += 1
+
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:135]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+    def plot_orders_satisfaction(self):
+        """
+        Plot the Orders Satisfaction section of the dashboard.
+        """
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:139]: Add return to docstring

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+            x="date",
+            y="Rate",
+            title="Orders Satisfaction",
+            marking_lines=[0, 1, 2, 3, 4, 5],
+            range_colors=["green", "yellow", "orange", "red", "purple", "maroon"],
+            x_axis_name="Date",
+            y_axis_name="Rate",
+        )
+        self.order += 1
+
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:159]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+    def plot_revenue_indicators(self):
+        """
+        Plot the Revenue Indicators section of the dashboard.
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:162]: Add return to docstring

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+                "align": "center",
+                "value": f"{round(real_revenue, 0)}€",
+                "color": "success",
+            },
+        ]
+        self.shimoku.plt.indicator(
+            cols_size=6, data=indicator_data, order=self.order, color="color"
+        )
+        self.order += len(indicator_data)
+
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:192]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+    def plot_orders_returned_orders(self):
+        """
+        Plot the Orders & Returned Orders section of the dashboard.
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:195]: Add return to docstring

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 12]
```diff
@@ -0,0 +1,250 @@
...
+        self.shimoku.plt.line(
+            title="Orders & Returned Orders",
+            data=monthly_metrics_df[["Orders", "Returned Orders", "Month"]],
+            order=self.order,
+            x="Month",
+            rows_size=4,
+            cols_size=6,
+        )
+        self.order += 1
+
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:218]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 13]
```diff
@@ -0,0 +1,250 @@
...
+        self.shimoku.plt.stacked_bar(
+            cols_size=6,
+            rows_size=3,
+            data=monthly_metrics_df[["Real Revenue", "Revenue Lost", "Month"]],
+            x="Month",
+            order=self.order,
+            show_values=["Real Revenue"],
+            title="Revenue Analysis",
+        )
+        self.order += 1
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:250]: Add return

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 14]
```diff
@@ -0,0 +1,250 @@
...
+            order=self.order,
+            x="Month",
+            rows_size=4,
+            cols_size=6,
+        )
+        self.order += 1
+
+    def plot_revenue_analysis(self):
+        """
+        Plot the Revenue Analysis section of the dashboard.
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:221]: Add return to docstring

File: templates/customer_satisfaction_performance/utils/components.py [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+    margin-top: 5px;
+}
+.modal-article .ordered-list li {
+    margin-top: 5px;
+}
+"""
+
+
+def create_title_name_head(title: str, subtitle: str) -> str:
+    """
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/utils/components.py:15]: Translate docstring to english.

Comment (HectorIdme) [reply templates/customer_satisfaction_performance/utils/components.py:15]: Add strcuture of docstring, parameters and return

File: templates/customer_satisfaction_performance/utils/utils.py [hunk 16]
```diff
@@ -0,0 +1,174 @@
+import pandas as pd
+import os
+
+
+def get_data(file_names):
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/utils/utils.py:5]: Add type for parameter file_names

File: templates/customer_satisfaction_performance/utils/utils.py [hunk 17]
```diff
@@ -0,0 +1,174 @@
+import pandas as pd
+import os
+
+
+def get_data(file_names):
+    """
+    Loads multiple CSV files into a dictionary of pandas DataFrames.
+
+    Parameters:
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/utils/utils.py:9]: Modify to standar use or Parameters or Args for all docstring, dont combine them.

File: templates/customer_satisfaction_performance/utils/utils.py [hunk 18]
```diff
@@ -0,0 +1,174 @@
...
+    """
+    dict_dfs = dict()
+    for file_name in file_names:
+        df = pd.read_csv(file_name)
+        dict_dfs[os.path.splitext(os.path.basename(file_name))[0]] = df
+
+    return dict_dfs
+
+
+def process_sales_data(df):
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/utils/utils.py:23]: Add type for parameter df

File: templates/customer_satisfaction_performance/utils/utils.py [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+            "Monthly Orders with Returns": monthly_orders_with_returns.to_dict(),
+            "Monthly Revenue Lost": monthly_revenue_lost.to_dict(),
+            "Monthly Real Revenue": monthly_real_revenue.to_dict(),
+        },
+    }
+
+    return results
+
+
+def get_status(value):
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/utils/utils.py:142]: Add type for parameter value

File: templates/customer_satisfaction_performance/utils/utils.py [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+    """
+    if value < 33:
+        return "success"
+    elif 33 <= value <= 66:
+        return "warning"
+    else:
+        return "error"
+
+
+def get_column_name_by_value(data_dict, value_to_find):
```
Comment (HectorIdme) [inline templates/customer_satisfaction_performance/utils/utils.py:160]: Add types for parametes data_dict and value_to_find

**Review by HectorIdme — CHANGES_REQUESTED** [review]
There are errors to fix.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 1 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:59]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 2 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:46]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 3 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:62]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 4 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:121]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 5 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:124]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 7 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:150]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 8 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:172]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 9 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:175]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 10 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:208]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 11 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:211]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 12 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:237]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 13 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:271]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/utils/components.py (hunk 15 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/utils/components.py:15]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/utils/utils.py (hunk 17 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/3ce80e2a1474175120a69959817737223569620a Changes made.

File: templates/customer_satisfaction_performance/utils/utils.py (hunk 18 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/utils/utils.py (hunk 19 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/utils/utils.py (hunk 20 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/utils/utils.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/dcf245413e61eeffcfd609bc742b7574046ed3ad. Changes made.

File: templates/customer_satisfaction_performance/data/generate_customer_satisfaction_performance.py [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+
+total_data = 10000
+output_file = "../../customer_satisfaction_performance.csv"
+date_ini = datetime(2023, 1, 1)
+date_end = datetime.now() 
+total_customers = 150
+cost_range = (15.00, 55.00)
+benefit_factor_range = (1.3, 1.5)
+
+def generate_data():
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/data/generate_customer_satisfaction_performance.py:16]: Be specific about the output type

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 22]
```diff
@@ -0,0 +1,272 @@
...
+        )
+        self.order += 1
+        self.shimoku.plt.shimoku_gauge(
+            cols_size=3,
+            value=orders_with_returns,
+            order=self.order,
+            rows_size=1,
+            name="Orders with Returns",
+            is_percentage=True,
+            color=orders_with_returns_status,
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:117]: there is not a simgle comment in this section, improve this

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 23]
```diff
@@ -0,0 +1,272 @@
...
+            values="value",
+            names="status",
+            order=self.order,
+            rows_size=3,
+            cols_size=4,
+            title="Customer Satisfaction",
+        )
+        self.order += 1
+        return True
+
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:144]: no comments

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 24]
```diff
@@ -0,0 +1,272 @@
...
+            x="date",
+            y="Rate",
+            title="Orders Satisfaction",
+            marking_lines=[0, 1, 2, 3, 4, 5],
+            range_colors=["green", "yellow", "orange", "red", "purple", "maroon"],
+            x_axis_name="Date",
+            y_axis_name="Rate",
+        )
+        self.order += 1
+        return True
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:171]: no coments here

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 25]
```diff
@@ -0,0 +1,272 @@
...
+                "value": f"{round(real_revenue, 0)}€",
+                "color": "success",
+            },
+        ]
+        self.shimoku.plt.indicator(
+            cols_size=6, data=indicator_data, order=self.order, color="color"
+        )
+        self.order += len(indicator_data)
+        return True
+
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:208]: no comments here =(

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 26]
```diff
@@ -0,0 +1,272 @@
...
+        self.shimoku.plt.line(
+            title="Orders & Returned Orders",
+            data=monthly_metrics_df[["Orders", "Returned Orders", "Month"]],
+            order=self.order,
+            x="Month",
+            rows_size=4,
+            cols_size=6,
+        )
+        self.order += 1
+        return True
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:236]: no comments here =(

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py [hunk 27]
```diff
@@ -0,0 +1,272 @@
...
+        self.shimoku.plt.stacked_bar(
+            cols_size=6,
+            rows_size=3,
+            data=monthly_metrics_df[["Real Revenue", "Revenue Lost", "Month"]],
+            x="Month",
+            order=self.order,
+            show_values=["Real Revenue"],
+            title="Revenue Analysis",
+        )
+        self.order += 1
```
Comment (rotorrest) [inline templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py:271]: no comments here =(
its important to explain the pandas transformation

Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/63a27708da52f92b63a0c2a6815b95fce69b4d81 changes made. Deleted the dashboard link from README.md.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 26 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/63a27708da52f92b63a0c2a6815b95fce69b4d81 changes made. Deleted the dashboard link from README.md.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 25 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/63a27708da52f92b63a0c2a6815b95fce69b4d81 changes made. Deleted the dashboard link from README.md.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 24 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/63a27708da52f92b63a0c2a6815b95fce69b4d81 changes made. Deleted the dashboard link from README.md.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 23 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/63a27708da52f92b63a0c2a6815b95fce69b4d81 changes made. Deleted the dashboard link from README.md.

File: templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py (hunk 22 above)
Comment (arturolinares24) [reply templates/customer_satisfaction_performance/paths/customer_satisfaction_performance.py]: https://github.com/shimoku-tech/shimoku-app-templates/commit/63a27708da52f92b63a0c2a6815b95fce69b4d81 changes made. Deleted the dashboard link from README.md.

**Review by HectorIdme — APPROVED** [review]

**Review by rotorrest — APPROVED** [review]

**Pull request merged**